pcl_utilities/src/tests/simple_test_concatenate_point_cloud.cpp

```diff
@@ -0,0 +1,181 @@
+ *   the queue is full, all messages in are sent
+ *   to the `concatenate_point_cloud_service`.
+ *
+ * The concatenated point cloud is then published to the topic
+ *   `concatenate_point_cloud/cloud_concatenated`
+ *
+ * Usage:
+ *    `ros2 launch pcl_utilities test_concatenate_point_cloud.xml point_cloud_topic:=<POINT_CLOUD_TOPIC>`
+ */
+#include <chrono>   // std::chrono::seconds
+#include <cstddef>  // size_t
+#include <cstdint>  // int64_t
+#include <deque>
+#include <functional>  // std::bind, std::placeholders
+#include <ios>  // std::fixed, std::setprecision
+#include <memory>  // std::make_shared
+#include <sstream>  // std::stringstream
+#include <string>
+#include <utility>  // std::move
+
+#include "rclcpp/executors.hpp"
+#include "rclcpp/logger.hpp"
+#include "rclcpp/logging.hpp"
+#include "rclcpp/node.hpp"
+#include "rclcpp/publisher.hpp"
+#include "rclcpp/subscription.hpp"
+#include "rclcpp/wait_for_message.hpp"
+#include "rcl_interfaces/msg/integer_range.hpp"
+#include "rcl_interfaces/msg/parameter_descriptor.hpp"
+
+#include "sensor_msgs/msg/point_cloud2.hpp"
+
+#include "pcl_utility_msgs/srv/pcl_concatenate_point_cloud.hpp"
+
+using pcl_utility_msgs::srv::PCLConcatenatePointCloud;
+using sensor_msgs::msg::PointCloud2;
+
+constexpr size_t MAX_POINT_CLOUDS = 100U;
+constexpr std::chrono::seconds MAX_WAIT_TIME {1U};
+
+class TestConcatenatePointCloudNode: public rclcpp::Node
+{
+private:
+    rclcpp::Client<PCLConcatenatePointCloud>::SharedPtr
+      concatenate_point_cloud_client_;
+    rclcpp::Publisher<PointCloud2>::SharedPtr output_publisher_;
+    std::deque<PointCloud2> history_;
+    std::string camera_topic_;
+    size_t num_point_clouds_;
+
+public:
+  TestConcatenatePointCloudNode()
+  : rclcpp::Node("simple_test_concatenate_point_cloud")
+  {
+    std::string client_topic = declare_parameter<std::string>("node_client_name");
+    camera_topic_ = declare_parameter<std::string>("point_cloud_topic");
+
+    // rclcpp uses int64_t internally, use explicit cast to size_t
+    rcl_interfaces::msg::ParameterDescriptor param_constraints;
+    rcl_interfaces::msg::IntegerRange integer_range;
+    integer_range.from_value = 0U;
+    integer_range.to_value = MAX_POINT_CLOUDS;
+    param_constraints.integer_range.push_back(integer_range);
+
+    num_point_clouds_ = static_cast<size_t>(
+      declare_parameter<int64_t>("num_point_clouds", std::move(param_constraints)));
+
+    concatenate_point_cloud_client_ =
+      create_client<PCLConcatenatePointCloud>(client_topic);
+
+    output_publisher_ = create_publisher<PointCloud2>(
+      "concatenate_point_cloud/cloud_concatenated", 1);
+
+    history_ = std::deque<PointCloud2>();
+  }
+
+  void spin()
+  {
+    while(rclcpp::ok())
+    {
+      PointCloud2 point_cloud_message;
+      bool was_retrieved = rclcpp::wait_for_message(
+        point_cloud_message, shared_from_this(),
+        camera_topic_, MAX_WAIT_TIME);
+
+      if (!was_retrieved)
+      {
```

Comment:
Brace should cuddle on if: https://docs.ros.org/en/humble/The-ROS2-Project/Contributing/Code-Style-Language-Versions.html#open-versus-cuddled-braces